File: src/main/java/com/github/kindrat/cassandra/client/ui/window/editor/main/table/DdlTextArea.java
package com.github.kindrat.cassandra.client.ui.window.editor.main.table;

import com.datastax.driver.core.TableMetadata;
import com.github.kindrat.cassandra.client.util.UIUtil;
import javafx.scene.control.TextArea;
import org.apache.commons.lang3.exception.ExceptionUtils;

import static com.github.kindrat.cassandra.client.util.CqlUtil.formatDDL;

public class DdlTextArea extends TextArea {
    public DdlTextArea() {
        setEditable(false);
        setPrefSize(200, 200);
        setVisible(false);
        UIUtil.fillParent(this);
    }

    public void showTableDDL(TableMetadata tableMetadata) {
        setText(formatDDL(tableMetadata.toString()));
        setVisible(true);
    }

    public void showException(Throwable throwable) {
        setText(ExceptionUtils.getStackTrace(throwable));
        setVisible(true);
    }
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/window/editor/main/table/DdlTextArea.kt
package com.github.kindrat.cassandra.client.ui.window.editor.main.table

import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata
import com.github.kindrat.cassandra.client.util.UIUtil.fillParent
import javafx.scene.control.TextArea
import org.apache.commons.lang3.exception.ExceptionUtils

class DdlTextArea : TextArea() {
    fun showTableDDL(tableMetadata: TableMetadata) {
//        setText(formatDDL(tableMetadata.describeWithChildren(true)));
        text = tableMetadata.describeWithChildren(true)
        isVisible = true
    }

    fun showException(throwable: Throwable?) {
        text = ExceptionUtils.getStackTrace(throwable)
        isVisible = true
    }

    init {
        isEditable = false
        setPrefSize(200.0, 200.0)
        isVisible = false
        fillParent(this)
    }
}